valkey_glide.stub.php

```
@@ -4120,6 +4123,88 @@ public function zunion(array $keys, ?array $weights = null, ?array $options = nu
+     *     'metrics' => [
+     *         'endpoint' => 'http://localhost:4318/v1/metrics'
+     *     ],
+     *     'flush_interval_ms' => 5000
+     * ]);
+     */
+    public static function initOpenTelemetry(array $config): bool;
+
+    /**
+     * Check if OpenTelemetry is initialized.
+     *
+     * @return bool True if OpenTelemetry is initialized, false otherwise
+     */
+    public static function isOpenTelemetryInitialized(): bool;
+
+    /**
+     * Get the current sample percentage for traces.
+     *
+     * @return int|null The sample percentage (0-100) if traces are configured, null otherwise
+     */
+    public static function getOpenTelemetrySamplePercentage(): ?int;
+
+    /**
+     * Set the percentage of requests to be sampled and traced at runtime.
+     * This allows dynamic adjustment of sampling without reinitializing OpenTelemetry.
+     *
+     * @param int $percentage The sample percentage (0-100)
+     * @return bool True if the percentage was set successfully
+     * @throws ValkeyGlideException If OpenTelemetry is not initialized or percentage is invalid
+     */
+    public static function setOpenTelemetrySamplePercentage(int $percentage): bool;
+
+    /**
+     * Determine if the current request should be sampled for OpenTelemetry tracing.
+     * Uses the configured sample percentage to randomly decide whether to create a span.
+     *
+     * @return bool True if the request should be sampled, false otherwise
+     */
+    public static function shouldSample(): bool;
+
+    /**
+     * Create a named OpenTelemetry span for custom tracing.
+     * The span must be manually ended using endOpenTelemetrySpan().
+     *
+     * @param string $name The name of the span
+     * @return int|null Span pointer on success, null if OpenTelemetry is not initialized or span creation failed
+     * @throws ValkeyGlideException If span name is invalid
+     */
+    public static function createOpenTelemetrySpan(string $name): ?int;
```

Should this throw instead of return null if OpenTelemetry isn't initialized or span creation failed?